packages/desktop-client/src/components/mobile/transactions/TransactionList.tsx

```diff
@@ -288,6 +303,58 @@ function SelectedTransactionsFloatingActionBar({
     };
   }, [dispatch]);
 
+  const canBeTransfer = useMemo(() => {
+    // only two selected
+    if (selectedTransactionsArray.length !== 2) {
+      return false;
+    }
+    const fromTrans = transactions.find(
+      t => t.id === selectedTransactionsArray[0],
+    );
+    const toTrans = transactions.find(
+      t => t.id === selectedTransactionsArray[1],
+    );
+
+    // previously selected transactions aren't always present in current transaction list
+    if (!fromTrans || !toTrans) {
+      return false;
+    }
+
+    return validForTransfer(fromTrans, toTrans);
+  }, [selectedTransactionsArray, transactions]);
+
+  const moreOptionsMenuItems: MenuItem<string>[] = [
+    {
+      name: 'duplicate',
+      text: t('Duplicate'),
+    },
+  ];
+
+  if (allTransactionsAreLinked) {
+    moreOptionsMenuItems.push({
+      name: 'unlink-schedule',
+      text: t('Unlink schedule'),
+    });
+  } else {
+    moreOptionsMenuItems.push({
+      name: 'link-schedule',
+      text: t('Link schedule'),
+    });
+  }
+
+  if (showMakeTransfer) {
+    moreOptionsMenuItems.push({
+      name: 'transfer',
+      text: t('Make transfer'),
+      disabled: !canBeTransfer,
+    });
+  }
+
+  moreOptionsMenuItems.push({
+    name: 'delete',
+    text: t('Delete'),
+  });
+
```

Comment:
```suggestion
  const moreOptionsMenuItems: MenuItem<string>[] = [
    { name: 'duplicate', text: t('Duplicate') },
    {
      name: allTransactionsAreLinked ? 'unlink-schedule' : 'link-schedule',
      text: t(allTransactionsAreLinked ? 'Unlink schedule' : 'Link schedule'),
    },
    ...(showMakeTransfer
      ? [
          {
            name: 'transfer',
            text: t('Make transfer'),
            disabled: !canBeTransfer,
          },
        ]
      : []),
    { name: 'delete', text: t('Delete') },
  ];
```

Reply:
I'll take the ternary (great shout, thanks!), but I'm not a massive fan of creating fake/temporary arrays just to unwrap them again... I'll push a modified version, hopefully this is okay!

Reply:
That works, I prefer the wrap/unwrap because it's more obvious at a glance what order the menu items will be in instead of the splice call but it's 6 and 2 threes really.